File: fizzbuzz.pl
#!/usr/bin/perl

use strict;
use warnings;

my $i = 1;

while ( $i <= 100 ) {

    my $f = ( $i % 3 == 0 ) ? 1 : 0;
    my $b = ( $i % 5 == 0 ) ? 1 : 0;

    if ( $f || $b ) {

        printf "%s%s\n",
            ( $f ) ? "Fizz" : "",
            ( $b ) ? "Buzz" : ""
        ; 
    }
    else {

        print "$i\n";
    }

    $i++;
}

File: fizzbuzz.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {

    int i;

    for (i=1; i<=100; i++) { 

        bool f = ( i % 3 == 0 ) ? true : false;
        bool b = ( i % 5 == 0 ) ? true : false;

        string result = ( f || b ) ? "" : to_string(i);

        result += ( f ) ? "Fizz" : "";
        result += ( b ) ? "Buzz" : "";

        cout << result << endl;
    }

    return 0;
}
